pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -0,0 +1,212 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package infraconfigurators
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vishvananda/netlink"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+)
+
+type Option func(handler *netdriver.MockNetworkHandler)
+
+func newMockedBridgeConfigurator(
+	vmi *v1.VirtualMachineInstance,
+	iface *v1.Interface,
+	handler *netdriver.MockNetworkHandler,
+	bridgeIfaceName string,
+	launcherPID int,
+	options ...Option) *BridgePodNetworkConfigurator {
+	configurator := NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+	for _, option := range options {
+		option(handler)
+	}
+	return configurator
+}
+
+var _ = Describe("Bridge infrastructure configurator", func() {
+	var (
+		ctrl    *gomock.Controller
+		handler *netdriver.MockNetworkHandler
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		handler = netdriver.NewMockNetworkHandler(ctrl)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	const (
+		bridgeIfaceName = "br0"
+	)
+
+	newVMIWithBridgeInterface := func(namespace string, name string) *v1.VirtualMachineInstance {
+		vmi := v1.NewMinimalVMIWithNS(namespace, name)
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+		v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+		return vmi
+	}
+
+	Context("discover link information", func() {
+		const (
+			ifaceName   = "eth0"
+			launcherPID = 1000
+		)
+
+		var (
+			defaultGwRoute netlink.Route
+			iface          *v1.Interface
+			podLink        *netlink.GenericLink
+			podIP          netlink.Addr
+			vmi            *v1.VirtualMachineInstance
+		)
+
+		BeforeEach(func() {
+			iface = v1.DefaultBridgeNetworkInterface()
+			vmi = newVMIWithBridgeInterface("default", "vm1")
+			podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: 1000}}
+			podIP = netlink.Addr{IPNet: &net.IPNet{IP: net.IPv4(10, 35, 0, 6), Mask: net.CIDRMask(24, 32)}}
+			defaultGwRoute = netlink.Route{Dst: nil, Gw: net.IPv4(10, 35, 0, 1)}
+		})
+
+		It("succeeds reading the pod link, pod IP, and routes", func() {
+			bridgeConfigurator := newMockedBridgeConfigurator(
+				vmi,
+				iface,
+				handler,
+				bridgeIfaceName,
+				launcherPID,
+				withLink(podLink),
+				withIPOnLink(podLink, podIP),
+				withRoutesOnLink(podLink, defaultGwRoute))
+			Expect(bridgeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(Succeed())
+			Expect(bridgeConfigurator.podNicLink).To(Equal(podLink))
+			Expect(bridgeConfigurator.podIfaceIP).To(Equal(podIP))
+			Expect(bridgeConfigurator.podIfaceRoutes).To(ConsistOf(defaultGwRoute))
```

Comment:
I remember commenting about it somewhere, but cannot find it now.
It would have been more complete and easier to troubleshoot if the assert would check the whole state of `bridgeConfigurator`. (i.e. asserting against an "expectedBridgeConfigurator", like it was done with the DHCP checks)

This is repeating at several places in these tests.

This is not a must, but it will be a nice thing to have.